tests/test-cases/Group1-Docker-Commands/1-43-Docker-CP.md

```diff
@@ -0,0 +1,80 @@
+20. Issue docker cp online:/newdir/test.txt foo.txt to the new VIC appliance
+21. Verify that test.txt is copied over to host successfully and overwrites foo.txt
+22. Issue docker cp foo.txt online:/doesnotexist/ to the new VIC appliance
+23. Issue docker cp foo.txt online:/ to the new VIC appliance
+24. Issue docker cp bar online:/ to the new VIC appliance
+25. Issue docker exec online ls / to the new VIC appliance to verify the copy operation succeeded
+26. Issue docker rm -f online to the new VIC appliance to remove online
+27. Issue docker create -i --name online -v vol1:/vol1 ${busybox} to the new VIC appliance
+28. Start online and issue docker cp bar online:/vol1/ to the new VIC appliance
+29. Issue docker exec online ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+30. Issue docker rm -f online to the new VIC appliance to clean up online
+31. Issue docker create -i --name offline -v vol1:/vol1 ${busybox} to the new VIC appliance
+32. Issue docker cp foo.txt offline:/vol1 to the new VIC appliance
+33. Start offline and issue docker exec offline ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+34. Issue docker rm -f offline to the new VIC appliance to clean up offline
+35. Issue docker create -i --name offline -v vol1:/vol1 -v vol2:/vol1/vol2 ${busybox} to the new VIC appliance
+36. Issue docker cp foo.txt offline:/vol1/vol2 to the new VIC appliance
+37. Start offline and issue docker exec offline ls /vol1/vol2 to the new VIC appliance to verify the copy operation succeeded
+38. Issue docker rm -f offline to the new VIC appliance to clean up offline
+39. Issue docker create -i --name offline -v vol1:/vol1 -v vol2:/vol1/vol2 -v vol3:/vol1/vol2/vol3 ${busybox} to the new VIC appliance
+40. Issue docker cp foo.txt offline:/vol1/vol2/vol3 to the new VIC appliance
+41. Start offline and issue docker exec offline ls /vol1/vol2/vol3 to the new VIC appliance to verify the copy operation succeeded
+42. Issue docker rm -f offline to the new VIC appliance to clean up offline
+43. Issue docker create -i --name online -v vol1:/vol1 ${busybox} to the new VIC appliance
+44. Issue docker create -i --name offline -v vol1:/vol1 ${busybox} to the new VIC appliance
+45. Start online and issue docker cp content offline:/vol1 to the new VIC appliance
+46. Issue docker exec online ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+47. Issue docker cp offline:/vol1 . to the new VIC appliance
+48. Verify that all contents in /vol1 are copied over successfully
+49. Issue docker rm -f offline to the new VIC appliance to clean up offline
+50. Generate a file of 4 MiB and issue docker cp largefile.txt online:/vol1/ to the new VIC appliance
+51. Issue docker exec online ls -l /vol1/ to the new VIC appliance to verify the copy operation succeeded
+52. Issue docker rm -f online to the new VIC appliance to clean up online
+53. Clean up files and directories and volumes created at setup time
```

Comment:
It might be easier to read if you lumped the failure cases together, but I don't see:
1. copy non-existent file out of offline or online
2. copy non-existent directory out of offline or online
3. copy non-existent directory into offline or online
4. And we should make sure that docker cp works with nfs volumes as well